src/api/Item.js

```diff
@@ -300,23 +305,46 @@ class Item extends Base {
             return Promise.reject();
         }
 
-        this.id = id;
-        this.successCallback = successCallback;
-        this.errorCallback = errorCallback;
+        const cache: APICache = this.getCache();
+        const key: string = this.getCacheKey(id);
+        const isCached: boolean = !options.forceFetch && cache.has(key);
 
-        // We use the parent folder's auth token since use case involves
-        // only content explorer or picker which works onf folder tokens
-        return this.xhr
-            .put({
+        if (isCached) {
+            return successCallback(cache.get(key));
+        }
+
+        try {
+            // We use the parent folder's auth token since use case involves
+            // only content explorer or picker which works on folder tokens
+            this.id = id;
+            this.successCallback = successCallback;
+            this.errorCallback = errorCallback;
+            const { fields } = options;
+
+            const fetchData: FetchData = {
                 url: this.getUrl(this.id),
                 data: {
                     shared_link: access === ACCESS_NONE ? null : { access },
                 },
-            })
-            .then(this.shareSuccessHandler)
-            .catch((e: $AxiosError<any>) => {
-                this.errorHandler(e);
-            });
+            };
+            if (fields) {
+                fetchData.params = { fields: fields.toString() };
+            }
+
+            const { data } = await this.xhr.put(fetchData);
+
+            const dataWithMissingFields = fillMissingProperties(data, fields);
+
+            if (cache.has(key)) {
+                cache.merge(key, dataWithMissingFields);
```

Comment:
It looks like `this.shareSuccessHandler` already contains some cache merging logic. Can we simplify the logic by moving it to that method?